FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -17,22 +17,24 @@ import { v4 } from "uuid";
 
 const DATABASE_FOLDER_NAME = "users";
 const POSTS_COLLECTION = "Posts";
+const POST_CATEGORIES_FOLDER_NAME = "postCategories"
 async function addUser(
   userID,
-  firstName,
-  lastName,
+  name,
   userName,
   signUpEmail,
-  selectedCategories
+  selectedCategories,
+  userLocation
 ) {
   const userDocRef = doc(database, DATABASE_FOLDER_NAME, userID);
   return await setDoc(userDocRef, {
     userID: userID,
-    FirstName: firstName,
-    LastName: lastName,
+    Name: name,
     UserName: userName,
     Email: signUpEmail,
+    selectedCategories: selectedCategories || [],
```

Comment:
Prefer this syntax. `??` will check whether `selectedCategories` is `null` or `undefined`.
When using `||`, any value that got evaluated as `true` would be returned.

```suggestion
    selectedCategories: selectedCategories ?? [],
```